starlight/views.py

```diff
@@ -6,23 +6,70 @@
 # License, or (at your option) any later version.
 
 from django.contrib.auth.views import logout
+from django.http import HttpResponseRedirect
 from django.shortcuts import render
 from django.views.decorators.csrf import csrf_protect
 
-from starlight.models import Skill, Employee
+from starlight.forms import EditForm, CompetencyForm
+from starlight.models import Skill, Employee, Competency
 
 
 def home(request):
     skills = Skill.objects.order_by('name')
     employees = Employee.objects.all()
-    return render(request, 'views/home.html', {'skills': skills, 'employees': employees, 'viewname': 'home'})
+    return render(request, 'views/home.html', {'skills': skills, 'employees': employees, 'viewgroup': 'home'})
 
 
 @csrf_protect
 def login(request):
     return render(request, 'login/login.html')
 
 
+def profile(request, id):
+    employee = Employee.objects.get(pk=id)
+    return render(request, 'profile_views/profile.html', {'employee': employee, 'viewgroup': 'profile'})
+
+
+def edit_competency(request, employee, id):
+    competency = Competency.objects.get(pk=id)
+    employee = Employee.objects.get(pk=employee)
+    skill = competency.skill
+    if request.method == 'POST':
+        form = EditForm(request.POST)
+        if form.is_valid():
+            interest = form.cleaned_data['interest']
+            experience = form.cleaned_data['experience']
+            competency_new, created = Competency.objects.get_or_create(skill=skill, interest=interest, experience=experience)
+            employee.competencies.remove(competency)
```

Comment:
To remove load on the DB, we can have these `.remove()` and `.add()` lines be conditional on if `created` is `True` ?